helix-term/src/ui/editor.rs

```diff
@@ -231,7 +231,45 @@ impl EditorView {
             }
         }
 
-        // render selections
+        // render gutters
+
+        let linenr: Style = theme.get("ui.linenr");
+        let warning: Style = theme.get("warning");
+        let error: Style = theme.get("error");
+        let info: Style = theme.get("info");
+        let hint: Style = theme.get("hint");
+
+        for (i, line) in (view.first_line..last_line).enumerate() {
+            use helix_core::diagnostic::Severity;
+            if let Some(diagnostic) = doc.diagnostics().iter().find(|d| d.line == line) {
+                surface.set_stringn(
+                    viewport.x - OFFSET,
+                    viewport.y + i as u16,
+                    "●",
+                    1,
+                    match diagnostic.severity {
+                        Some(Severity::Error) => error,
+                        Some(Severity::Warning) | None => warning,
+                        Some(Severity::Info) => info,
+                        Some(Severity::Hint) => hint,
+                    },
+                );
+            }
+
+            // line numbers having selections are rendered differently
+            surface.set_stringn(
+                viewport.x + 1 - OFFSET,
+                viewport.y + i as u16,
+                format!("{:>5}", line + 1),
+                5,
+                linenr,
+            );
+        }
+
+        // render selections and selected linenr(s)
+        let linenr_select: Style = theme
+            .try_get("ui.linenr.selected")
+            .unwrap_or_else(|| theme.get("ui.linenr"));
```

Comment:
Can we add a comment to say when we deprecate this fallback? Since we are new we don't have to keep this forever, maybe we can show a warning and then later after few version we can remove this.

Reply:
I think this is a reasonable fallback because not all themes want to define a `.selected` version. I don't think we'll remove this

Reply:
I think we should document a minimum a themes that are necessary for helix to work. Maybe we can do a check at startup or we just do fallback for all of them?

Reply:
Almost everything can just fall back to `ui.text`, it just currently doesn't (it uses `#0000FF`) so that if you forget a key it stands out.